python/pyarrow/io.pxi

```diff
@@ -1377,6 +1377,21 @@ cdef shared_ptr[CBuffer] as_c_buffer(object o) except *:
     return buf
 
 
+cdef NativeFile wrap_python_file(object source, str mode):
```

Comment:
Is there a reason not to reuse `_get_native_file`?

Reply:
I explicitly wanted to avoid the path/fs case and address only file-like objects. `_get_native_file` will probably work, I'll try replacing it